examples, docs: Harmonize --ble-controller in all examples and documentation

enkiusz opened the pull request:
#### Testing

Manual test of new help text:

```
$ out/linux-x64-chip-tool/chip-tool pairing ble-wifi --help
[…]
[--ble-adapter]:
  Bluetooth adapter index. 0 -> hci0, 1 -> hci1, etc.
```

```
$ out/linux-x64-camera-controller/chip-camera-controller --help
[…]
[--ble-adapter]:
  Bluetooth adapter index. 0 -> hci0, 1 -> hci1, etc.
```

```
$ out/linux-x64-fabric-admin-rpc/fabric-admin --help  
[…]
[--ble-adapter]:
  Bluetooth adapter index. 0 -> hci0, 1 -> hci1, etc.
```

enkiusz committed:
examples: Add help for the --ble-adapter option

Signed-off-by: Maciej Grela <m.grela@samsung.com>

andy31415 reviewed (changes requested):
CI does not verify `--help`. Please add description on how you verified that --help worked (i.e. manually running it and expected output)

enkiusz commented:
I have improved the PR description.

arkq commented on examples/camera-controller/commands/common/CHIPCommand.h:
```diff
@@ -93,7 +93,7 @@ class CHIPCommand : public Command
 #endif // CHIP_CONFIG_TRANSPORT_TRACE_ENABLED
         AddArgument("trace-to", &mTraceTo, "Trace destinations, comma-separated (" SUPPORTED_COMMAND_LINE_TRACING_TARGETS ")");
-        AddArgument("ble-adapter", 0, UINT16_MAX, &mBleAdapterId);
+        AddArgument("ble-adapter", 0, UINT16_MAX, &mBleAdapterId, "Bluetooth adapter index. 0 -> hci0, 1 -> hci1, etc.");
```
I'm not sure whether `hciX` is a universal nomenclature. It works for Linux, but how about mac OS? Also, add info that `0` is the default.

Also, there is one inconsistency in examples. Here we have `--ble-adapter` and other examples (mostly apps) have `--ble-device`. Maybe it would be nice to unify it? I would opt for `--ble-adapter` since we are configuring adapter, not a remote BLE device connected to it.

enkiusz replied:
I agree that `--ble-adapter` is more descriptive than a generic "device" term. 
There is a third possibility `--ble-controller` which could be a better term consistent with the Bluetooth nomenclature. After all HCI is the Host-Controller Interface. 

I will create a poll on the slack channel to gather some feedback on what we would prefer as a group.

bzbarsky-apple commented on examples/chip-tool/commands/common/CHIPCommand.h:
```diff
@@ -96,7 +96,7 @@ class CHIPCommand : public Command
 #endif // CHIP_CONFIG_TRANSPORT_TRACE_ENABLED
         AddArgument("trace-to", &mTraceTo, "Trace destinations, comma-separated (" SUPPORTED_COMMAND_LINE_TRACING_TARGETS ")");
-        AddArgument("ble-adapter", 0, UINT16_MAX, &mBleAdapterId);
+        AddArgument("ble-adapter", 0, UINT16_MAX, &mBleAdapterId, "Bluetooth adapter index. 0 -> hci0, 1 -> hci1, etc.");
```
The exact names depend on the operating system and whatnot, right?

enkiusz replied:
Yes, I have pushed a new version which references the docs for details on the controller naming in particular platforms. I wrote such a description for Linux as a start.

enkiusz committed:
examples, docs: Harmonize --ble-controller in all examples and documentation

- create a platform/linux tree for Linux-specific BLE selection logic, very simple but simple to extend later and add other platforms
- reformatted list depth "Running the Complete Example" sections on most Linux as it did not make sense

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by prettier-markdown

enkiusz committed:
python: Additional places where --ble-interface-id lingered

Signed-off-by: Maciej Grela <m.grela@samsung.com>
